src/visualization/makie_visualization.jl

```diff
@@ -0,0 +1,263 @@
+
+Generates a plotting triangulation which can be used by Makie's `mesh` function.
+
+Arguments:
+- sol: TrixiODESolution
+- variable_to_plot: index of solution field to plot
+
+Example:
+```julia
+tri = generate_plotting_triangulation(sol,1)
+Makie.mesh(tri,color=getindex.(tri.position,3))
+```
+The color is specified to be the z-value of the solution, which there does not seem to be a default solution for yet.
+"""
+function generate_plotting_triangulation(sol::TrixiODESolution, variable_to_plot::Int; nvisnodes=10)
+
+    semi = sol.prob.p
+    dg = semi.solver
+    @unpack equations, cache, mesh = semi
+
+    # make solution
+    u = sol.u[end]
+    u = wrap_array(u, mesh, equations, dg, cache)
+
+    num_nodes_1D = length(dg.basis.nodes)
+    num_nodes = num_nodes_1D^2
+    num_elements = nelements(dg, cache)
+
+    # build nodes on reference element (seems to be the right ordering)
+    r1D = dg.basis.nodes
+    r = vec([r1D[i] for i = 1:num_nodes_1D, j = 1:num_nodes_1D])
+    s = vec([r1D[j] for i = 1:num_nodes_1D, j = 1:num_nodes_1D])
+
+    # reference plotting nodes
+    Vp1D = plotting_interpolation_matrix(dg; nvisnodes=nvisnodes)
+    Vp = kron(Vp1D, Vp1D)
+    n_plot_nodes = size(Vp, 1)
+
+    # create triangulation for plotting nodes
+    rp,sp = (x->Vp*x).((r, s)) # interpolate dg nodes to plotting nodes
+
+    # construct a triangulation of the plotting nodes
+    t = permutedims(plotting_triangulation((rp, sp)))
+    makie_triangles = Makie.to_triangles(t)
+
+    # trimesh[i] holds the GeometryBasics.Mesh containing solution plotting
+    # information on the ith element.
+    trimesh = Vector{GeometryBasics.Mesh{3,Float32}}(undef, num_elements)
+    coordinates_tmp = zeros(Float32, num_nodes, 3)
+    coordinates = zeros(Float32, n_plot_nodes, 3)
+    for element in eachelement(dg, cache)
+
+        # extract x,y coordinates and solutions on each element
+        sk = 1
+        for j in eachnode(dg), i in eachnode(dg)
+            u_node = get_node_vars(u, equations, dg, i, j, element)
+            xy_node = get_node_coords(cache.elements.node_coordinates, equations, dg, i, j, element)
+            coordinates_tmp[sk,1:2] .= xy_node
+            coordinates_tmp[sk,3] = u_node[variable_to_plot]
+            sk += 1
+        end
+
+        # interpolates both xy coordinates and solution values to plotting points
+        mul!(coordinates, Vp, coordinates_tmp)
+
+        trimesh[element] = GeometryBasics.normal_mesh(Makie.to_vertices(coordinates), makie_triangles)
+    end
+
+    # merge meshes on each element into one large mesh
+    plotting_mesh = merge([trimesh...])
+    return plotting_mesh
+end
+
+function generate_plotting_wireframe(sol::TrixiODESolution, variable_to_plot::Int; nvisnodes=5)
+
+    semi = sol.prob.p
+    @unpack equations, mesh, cache = semi
+    dg = semi.solver
+
+    num_nodes_1D = length(dg.basis.nodes)
+    num_nodes = num_nodes_1D^2
```

Comment:
I guess this `2` here is something like `ndims(mesh)`? If so, could you please use that to make it more clear?